### PR TITLE
feat(bpmn): project interpreter diagnostics onto the scope's execution log

### LINKED ISSUE #8043: BPMN W13: project interpreter diagnostics into the workflow journal, keyed by element id
Part of #7909 (W13, core half). The Studio half is filed in elsa-studio. Small, and it unblocks the only thing standing between the runtime and an instance viewer that shows where a token is.

## Why

Under Option A only bound work has an activity id, so Studio's stats channel (`UpdateActivityStatsAsync(activityId, …)`) lights up tasks and nothing else. Gateways, events and flows are decisions the interpreter made, and it records every one of them in `BpmnExecutionState.Diagnostics` as a `bpmnDiagnosticEvent` `{diagnosticId, kind, message, elementId, flowId, tokenId, details}`. D1 says: project those into the journal, keyed by element id, and let Studio build its overlay from there. The runtime activity tree stays untouched and the projection is write-only.

## Where

`src/modules/Elsa.Bpmn/Hosting/BpmnCommandApplier.cs`, or wherever the evaluation result is handled before `Prune()` runs (state is staged through `BpmnScopeMemory`). Today the applier writes exactly one journal entry, `"Torn down by BPMN"` in `BpmnWorkTeardown.cs`.

## Shape

- After each evaluation and **before** pruning, emit one `AddExecutionLogEntry(eventName, message, source, payload)` on the scope context per new diagnostic. Diagnostics are capped at 200 and pruned, so projecting from persisted state later loses entries; project at evaluation time and track the last projected `diagnosticId` (or use `sequence`) in scope memory so a resumed scope does not re-emit.
- Event name: one stable name per `kind` (follow the existing convention of short past-tense names such as `Faulted`, `Suspended`), with `source` `"BPMN"`. Payload carries `elementId`, `flowId`, `tokenId`, `kind` and `details` verbatim. Studio keys on `elementId`; do not fold it into the message.
- Do not emit for the scope's own start and completion; those are already journaled as the activity's own lifecycle.

## Traps

- Volume. A long-running scope evaluates many times; 200 entries per evaluation is the cap, not the norm, but a tight loop can flood the journal. Measure with the persistence test's multi-iteration loop and decide on a per-kind filter if it is unreasonable.
- Emit on the scope context, not the child's. The diagnostic describes the scope's decision about a child, and the child may already be torn down.

## Verification

Integration test in `test/integration/Elsa.Bpmn.IntegrationTests`: run the parallel split-and-join process and assert the journal carries entries for the split gateway, both flows, and the join, each keyed by element id, and none is duplicated after a suspend and resume.

### PR DESCRIPTION
**Pull request by sfmskywalker**
Closes #8043. Part of #7909 (W13, core half). Unblocks elsa-workflows/elsa-studio#1000, the element-keyed instance overlay.

## What changed

`BpmnScopeHost` now projects each new interpreter diagnostic onto the scope's own execution log, so Studio can light up gateways, events and flows, not only bound tasks.

- Projection runs from one method, on every evaluation path, **before** `Prune()` caps diagnostics at 200.
- One `AddExecutionLogEntry` per new diagnostic, on the scope's context (never the child's, which may already be torn down), with `source = "BPMN"`.
- The event name is the library's `BpmnDiagnosticKind` member name. The names are published as constants in `BpmnDiagnosticEventNames`, and a reflection test pins them to the enum in both directions, so a library upgrade cannot drift them silently.
- The payload is the public record `BpmnDiagnosticLogPayload`: `diagnosticId`, `elementId`, `flowId`, `tokenId`, `kind` (the member name as a string, so Studio never depends on integer values) and `details`, verbatim.
- A persisted high-water mark (`BpmnDiagnosticsCursor`, stored beside the execution state) stops a resumed scope from re-emitting. Diagnostic ids are `diag:{n}` with a monotonic counter (verified against Bpmn.Semantics 0.2.0). An id that does not parse is skipped with a warning; the projection never faults a workflow.
- Documented in `doc/wiki/bpmn-workflows.md` as a compatibility surface.

## One deliberate reading of the spec

The issue says not to emit for "the scope's own start and completion". The projection skips exactly the scope-level diagnostics, meaning those that name neither an element nor a flow (the scope's `Completed` summary). A start event's token emission names the start event element and is projected, because the overlay should light up start events and it does not duplicate the activity's own lifecycle entry. An earlier draft excluded every flowless `TokenEmitted` and would have dropped error- and cancel-boundary catches; review caught it and a test now pins the boundary case.

## Volume

Measured with the 12-iteration sequential multi-instance loop in `BpmnPersistenceTests`: each iteration after the first adds two projected entries (`Consumed`, `Scheduled`), 106 diagnostics in total. No per-kind filter is applied; journal growth is proportional to the process's work, like any activity's journal.

## Verification

```
dotnet build Elsa.sln                                              # 0 errors
dotnet test test/integration/Elsa.Bpmn.IntegrationTests            # 57/57
dotnet test test/unit/Elsa.Bpmn.UnitTests                          # 20/20
dotnet test test/unit/Elsa.Bpmn.Interchange.UnitTests              # 8/8 (incl. the no-duplication guard)
dotnet test test/integration/Elsa.Bpmn.Interchange.IntegrationTests  # 57/57
```

New tests: the parallel split and join projects entries for the split gateway, both flows and the join, each keyed by element or flow id, with no duplicates across a real suspend, state round trip through the workflow state serializer, and resume; entries land on the scope's own context; an error-boundary catch projects its token emission; the scope-level `Completed` is not projected; the event-name constants match the enum in both directions.

Review: two iterations on the standards and spec axes; four must-fix findings resolved. Won't-fix, recorded: event names mirror the library enum 1:1 rather than being translated to past tense.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/modules/Elsa.Bpmn/Hosting/BpmnScopeHost.cs [hunk 1]
```diff
@@ -209,6 +217,70 @@ private async ValueTask ApplyAsync(BpmnScopeMemory memory, BpmnEvaluation evalua
...
+            BpmnScopeMemory.Write(_context, BpmnScopeMemory.DiagnosticsCursorPropertyKey, new BpmnDiagnosticsCursor(highWaterMark));
+    }
+
+    /// <summary>
+    /// The numeric ordinal in a diagnostic id (<c>diag:N</c>) — a pure function of the interpreter's own
+    /// mutation-order sequence, so it sorts the same as arrival order. Never throws: an id that does not match the
+    /// expected format fails to parse rather than faulting the evaluation, since this runs on every evaluation.
+    /// </summary>
+    private static bool TryGetDiagnosticSequence(string diagnosticId, out int sequence) =>
+        int.TryParse(diagnosticId.AsSpan(diagnosticId.IndexOf(':') + 1), out sequence);
```
[the remark this answers is not in the document]
Comment (sfmskywalker) [reply src/modules/Elsa.Bpmn/Hosting/BpmnScopeHost.cs]: Fixed in 2db10ede6. The parser now requires the exact ordinal prefix `diag:` and a non-negative integer suffix (`NumberStyles.None`, invariant culture). Anything else takes the warn-and-skip path without touching the cursor. Unit tests cover `999`, `foreign:999`, `diag:-1`, `diag:`, `diag:abc` and `diag: 5`.

File: src/modules/Elsa.Bpmn/Hosting/BpmnScopeHost.cs [hunk 2]
```diff
@@ -209,6 +217,70 @@ private async ValueTask ApplyAsync(BpmnScopeMemory memory, BpmnEvaluation evalua
...
+    /// this scope's decision about a child, and the child may already be torn down by the time this runs. Called
+    /// with the evaluation's own <see cref="BpmnEvaluation.State"/>, before <c>Prune()</c> caps
+    /// <see cref="BpmnExecutionState.Diagnostics"/> at 200 entries, because projecting from what was actually
+    /// persisted would lose whatever pruning already dropped. The last diagnostic id it has projected is kept in
+    /// <see cref="BpmnScopeMemory.DiagnosticsCursorPropertyKey"/> so a resumed scope does not re-emit one a
+    /// previous evaluation already turned into a journal entry.
+    /// </remarks>
+    private void ProjectDiagnostics(BpmnExecutionState state)
+    {
+        var lastProjectedSequence = BpmnScopeMemory.Read<BpmnDiagnosticsCursor>(_context, BpmnScopeMemory.DiagnosticsCursorPropertyKey)?.LastSequence ?? 0;
```
[the remark this answers is not in the document]
Comment (sfmskywalker) [reply src/modules/Elsa.Bpmn/Hosting/BpmnScopeHost.cs]: Fixed in 2db10ede6. When the cursor property is absent but the scope has previously persisted state, the cursor is seeded from the highest valid sequence in that prior state before projecting, so only this evaluation's diagnostics are journaled. A new scope still starts at zero. An integration test suspends a scope, removes the cursor from the persisted state, resumes, and asserts no pre-suspend diagnostic reappears; it fails without the fix.

File: src/modules/Elsa.Bpmn/Hosting/BpmnScopeHost.cs [hunk 3]
```diff
@@ -209,6 +220,115 @@ private async ValueTask ApplyAsync(BpmnScopeMemory memory, BpmnEvaluation evalua
...
+        var highest = 0;
+
+        if (priorState is null)
+            return highest;
+
+        foreach (var diagnostic in priorState.Diagnostics)
+        {
+            if (TryGetDiagnosticSequence(diagnostic.DiagnosticId, out var sequence) && sequence > highest)
+                highest = sequence;
+        }
```
[the remark this answers is not in the document]
Comment (sfmskywalker) [reply src/modules/Elsa.Bpmn/Hosting/BpmnScopeHost.cs:306]: Not taken. The loop filters and extracts in one step: `TryGetDiagnosticSequence` both decides whether an id is valid and yields its sequence through the out parameter. Splitting that into a `Where` plus a second parse in the body, as suggested, parses every id twice to satisfy the rule. Recorded as a won't-fix in the delivery report.
